File: frontend/src/types/scan.ts
export type ScanType =
  | "SAST"
  | "SCA"
  | "Secrets"
  | "Container Security"
  | "IaC"
  | "DAST";

export interface ScanHistoryItem {
  id: string;
  scan_type: ScanType;
  tool: string;
  filename: string;
  status: string;
  uploaded_at: string;
  findings: number;
}

export const SCAN_TYPES: ScanType[] = [
  "SAST",
  "SCA",
  "Secrets",
  "Container Security",
  "IaC",
  "DAST",
];

export interface ScanUploadResponse {
  scan_id: string;
  scan_type: ScanType;
  tool: string;
  filename: string;
  status: string;
  findings: number;
}

File: backend/app/schemas/scan.py
from enum import Enum
from datetime import datetime

from pydantic import BaseModel
from uuid import UUID


class ScanType(str, Enum):
    SAST = "SAST"
    SCA = "SCA"
    SECRETS = "Secrets"
    CONTAINER_SECURITY = "Container Security"
    IAC = "IaC"
    DAST = "DAST"


class ScanUploadResponse(BaseModel):
    scan_id: UUID
    scan_type: ScanType
    tool: str
    filename: str
    status: str
    findings: int


class ScanHistoryItem(BaseModel):
    id: UUID
    scan_type: ScanType
    tool: str
    filename: str
    status: str
    uploaded_at: datetime
    findings: int
